pkg/plugin/plugin.go

```diff
@@ -16,6 +16,33 @@ type Plugin struct{}
 
 var _ api.Plugin = &Plugin{}
 
+func (p *Plugin) MergeConfig(cs, oldCs *acsapi.ContainerService) {
```

Comment:
most of these are pointers, we're going to need to make a copy of the value rather than just assign the pointer.  Maybe it's worth having a look at https://github.com/imdario/mergo.  I believe Kubernetes uses it too.

Reply:
Hrm, two comments: 1. if the API allows passing pointers then it's assumed that the implementation can muck with the pointers, so I think it's a responsibility of the caller to ensure that it provides copies rather than the original pointers, 2. why would we need to copy at all?